File: src/main/scala/warmUp/Scala/PlusMinus.scala
/**
  * @author Ram Saran Vuppuluri
  *         Problem statement
  * @see https://www.hackerrank.com/challenges/plus-minus/problem
  */
package warmUp.Scala

import java.util.Scanner

object PlusMinus extends App {
  val scanner = new Scanner(System.in)
  val arraySize = scanner.nextInt()

  var negativeCount = 0
  var positiveCount = 0
  var zeroCount = 0

  for (i <- 1 to arraySize) {
    val nextInt = scanner.nextInt()
    if (nextInt > 0)
      positiveCount += 1
    if (nextInt < 0)
      negativeCount += 1
    if (nextInt == 0)
      zeroCount += 1
  }

  println((positiveCount * 1.0) / arraySize)
  println((negativeCount * 1.0) / arraySize)
  println((zeroCount * 1.0) / arraySize)
}

File: src/main/scala/warmUp/Java/PlusMinus.java
/**
 * @author Ram Saran Vuppuluri
 * Problem statement
 * @see
 */
package warmUp.Java;

import java.util.Scanner;

public class PlusMinus {
    public static void main(String... args) {
        Scanner sin = new Scanner(System.in);
        int arraySize = sin.nextInt();
        int positiveSum = 0;
        int negativeSum = 0;
        int zeroSum = 0;

        for (int i = 0; i < arraySize; i++) {
            int nextInt = sin.nextInt();
            if (nextInt > 0) {
                positiveSum++;
            } else if (nextInt < 0) {
                negativeSum++;
            } else {
                zeroSum++;
            }
        }

        System.out.println((positiveSum * 1.0) / arraySize);
        System.out.println((negativeSum * 1.0) / arraySize);
        System.out.println((zeroSum * 1.0) / arraySize);
    }
}
